### PR TITLE
Modifies the usage of service create, link and unlink by using separate files for Kubernetes resources

### LINKED ISSUE #4886: Use a separate files for Kubernetes resources
/kind user-story

## User Story
As an odo user, I want to have Kubernetes resources in a separate file to easily review them or use 3rd party tools to deploy them.

## Acceptance Criteria 
- [ ] `odo link` should put `ServiceBinding` into a separate file and use `uri` to reference it in `kubernetes` devfile component.
- [ ] `odo service create` should put service CR into a separate file and use `uri` to reference it in `kubernetes` devfile component.
- [ ] HTTPS URIs should be supported
- [ ] Default should be URI with an option to let user specify `--inlined`

## implementation notes:
- generated yaml files should be placed in a separate directory to avoid polluting project root. For example `./kubernetes/` by default. This directory should be configurable using global odo preference.
- filenames should use a pattern that is descriptive and avoids conflicts
- commands like  `odo service list` or `odo describe` needs to be able to handle a situation when `kubernetes` component is referenced using `uri` field. 

### example
```
odo service create  postgresql-operator.v0.1.1/Database -p databaseName=mydb -p databaseUser=user -p databasePassword=pass
```
Instead of inlining Operator CR into a devfile it will create `kubernetes` directory (if it doesn't already exist) and save Operator CR to `kubernetes/database.yaml`
Than it will add it as a `kubernetes` component into a devfile.yaml
``` 
....
- kubernetes:
    uri: kubernetes/database.yaml
...
```

/kind user-story

### PR DESCRIPTION
**Pull request by mik-dass**
**What type of PR is this?**

/kind feature

**What does this PR do / why we need it**:

It modifies the usage of service create, link and unlink by using separate files for Kubernetes resources.

**Which issue(s) this PR fixes**:

Fixes #4886 

**PR acceptance criteria**:

- [X] Unit test 

- [X] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

- TBA

### CODE REVIEW

**Comment by mik-dass** [comment]
```
✗  Failed to start component with name "opggji". Error: Failed to create the component: error while waiting for deployment rollout: timeout while waiting for opggji-app deployment roll out
```
/retest

**Comment by mik-dass** [comment]
```
✗  Failed to start component with name "smspfj". Error: Failed to create the component: error while waiting for deployment rollout: timeout while waiting for smspfj-app deployment roll out
```
/retest

File: pkg/service/service.go [hunk 1]
```diff
@@ -488,8 +528,57 @@ func AddKubernetesComponentToDevfile(crd, name string, devfileObj parser.Devfile
...
+	}
+
+	err = devfileObj.Data.AddComponents([]devfile.Component{{
+		Name: name,
+		ComponentUnion: devfile.ComponentUnion{
+			Kubernetes: &devfile.KubernetesComponent{
+				K8sLikeComponent: devfile.K8sLikeComponent{
+					BaseComponent: devfile.BaseComponent{},
+					K8sLikeComponentLocation: devfile.K8sLikeComponentLocation{
+						Uri: filepath.Join(name + ".yaml"),
```
Comment (feloy) [inline pkg/service/service.go:562]: I would expect to have the uriPath contained in this Uri field

**Comment by feloy** [comment]
@mik-dass Could you please delete the files in kubernetes when running odo delete --all ?

File: pkg/devfile/adapters/common/utils.go [hunk 2]
```diff
@@ -219,3 +221,37 @@ func GetSyncFilesFromAttributes(commandsMap PushCommandsMap) map[string]string {
...
+
+		if component.Openshift != nil && component.Openshift.Uri != "" {
+			uri = component.Openshift.Uri
+		}
+
+		if uri == "" {
+			continue
+		}
+
+		completePath := filepath.Join(componentContext, uri)
```
Comment (feloy) [inline pkg/devfile/adapters/common/utils.go:250]: You should test that the URI is a relative path, and not an http[s]:// URI

Comment (mik-dass) [reply pkg/devfile/adapters/common/utils.go:259]: Fixed

**Comment by mik-dass** [comment]
```
[ssh:Fedora-32-minikube] [odo] I0917 08:12:59.182670  607482 schema.go:46] validated devfile schema
[ssh:Fedora-32-minikube] [odo] I0917 08:12:59.182938  607482 validate.go:45] Successfully validated devfile sections
[ssh:Fedora-32-minikube] [odo]  ✗  open kubernetes: no such file or directory
```
/retest

**Comment by mik-dass** [comment]
> [odo] I0917 12:01:01.266849   70473 validate.go:45] Successfully validated devfile sections
[odo]  ✗  open /tmp/698150647/kubernetes: no such file or directory

/hold

**Comment by mik-dass** [comment]
```
 Registry DefaultDevfileRegistry is not set up properly with error: Get "https://registry.stage.devfile.io/index": connection error: PROTOCOL_ERROR, please check the registry URL and credential (refer `odo registry update --help`)
[ssh:Fedora-32-minikube] [odo] 
[ssh:Fedora-32-minikube] [odo] I0917 14:28:16.256517 1853624 util.go:483] path devfile.yaml doesn't exist, skipping it
[ssh:Fedora-32-minikube] [odo] Devfile Object Validation
[ssh:Fedora-32-minikube] [odo]  •  Checking devfile existence  ...
[ssh:Fedora-32-minikube] [odo]  ✗  Checking devfile existence [27741ns]
[ssh:Fedora-32-minikube] [odo]  ⚠  Devfile component type nodejs is not suppo
```
/retest

**Pull request merged**